### PR TITLE
feat(comp: spin):  add spin component

### PR DESCRIPTION
**Pull request by liuzaijiang**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IduxFE/components/blob/main/docs/contributing.zh.md#commit)
- [x] Tests for the changes have been added/updated or not needed
- [x] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Component style update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

**Comment by danranVm** [comment]
PR 的 title 不符合规范，
commit 需要做一下合并，使用 rebase 而不是 merge 操作
另外 lint 的 CI 挂掉了，还需要在本地执行一下 `npm run lint-fix` 看看是哪里不对

File: packages/components/spin/docs/index.zh.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## 何时使用
+
+当部分区域正在等待异步数据或在呈现过程中，让用户能更好感知数据请求和界面渲染，而且也能阻止用户在数据请求过程中对视图层有其他操作导致一些不必要的错误；
+
+## API
+
+### `ix-spin`
+
+CONFIGS
```
Comment (danranVm) [inline packages/components/spin/docs/index.zh.md:18]:
```suggestion
#### props
```

File: packages/components/spin/docs/index.zh.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+|参数 | 说明 |  类型  | 默认值 | 是否支持全局配置|
+| --- | --- | --- | --- | --- |
+|`spinning` | 是否显示加载遮罩层|`boolean`| `true` | `false`|
+| `icon`| 加载图标名称 | `string` | `loading`| `true`|
+| `tip`| 加载提示文字描述 |  `string`  | `''`| `true`|
+| `tipAlign`| 文字描述与加载图标的排列方式 | `string` | `vertical(vertical：文字排列在图标下方；horizontal： 文字排列在图标同水平方向的后方)`| `true`|
+|`wrapperClassName`| 组件最外包裹层class名 |  `string`  | `''`| `true`|
+| `size`| icon 和 tip大小 |  `string(small \| medium \| large)`  | `small`| `true`|
+
+SLOTS
```
Comment (danranVm) [inline packages/components/spin/docs/index.zh.md:29]:
```suggestion
#### slots
```

File: packages/components/spin/docs/index.zh.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+当部分区域正在等待异步数据或在呈现过程中，让用户能更好感知数据请求和界面渲染，而且也能阻止用户在数据请求过程中对视图层有其他操作导致一些不必要的错误；
+
+## API
+
+### `ix-spin`
+
+CONFIGS
+
+|参数 | 说明 |  类型  | 默认值 | 是否支持全局配置|
```
Comment (danranVm) [inline packages/components/spin/docs/index.zh.md:20]:
```suggestion
| 参数 | 说明 |  类型  | 全局配置 | 默认值 | 备注 |
```

File: packages/components/spin/docs/index.zh.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+| --- | --- | --- | --- | --- |
+|`spinning` | 是否显示加载遮罩层|`boolean`| `true` | `false`|
+| `icon`| 加载图标名称 | `string` | `loading`| `true`|
+| `tip`| 加载提示文字描述 |  `string`  | `''`| `true`|
+| `tipAlign`| 文字描述与加载图标的排列方式 | `string` | `vertical(vertical：文字排列在图标下方；horizontal： 文字排列在图标同水平方向的后方)`| `true`|
```
Comment (danranVm) [inline packages/components/spin/docs/index.zh.md:25]:
```suggestion
| `tipAlign`| 文字描述与加载图标的排列方式 | `vertical \| horizontal ` | `vertical`| - | vertical：文字排列在图标下方；horizontal： 文字排列在图标同水平方向的后方 |
```

File: packages/components/spin/src/Spin.vue [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      hasDefaultSlot.value = !!slots.default
+    })
+
+    const spinnerClassName = computed(() => {
+      const hasClass = props.spinning && tip.value
+      const size = props.size ?? spinConfig.size
+      return [
+        hasClass ? `ix-spin-spinner-tip-${props.tipAlign ?? spinConfig.tipAlign}` : '',
+        hasClass && size !== 'medium' ? `ix-spin-spinner-${size}` : '',
+      ]
```
Comment (danranVm) [inline packages/components/spin/src/Spin.vue:52]:
```suggestion
      if(!props.spinning  || !tip.value){
        return []
      } 
      const size = props.size ?? spinConfig.size
      const tipAlign =  props.tipAlign ?? spinConfig.tipAlign
      return [
        `ix-spin-spinner-tip-${tipAlign}` ,
        size !== 'medium' ? `ix-spin-spinner-${size}` : '',
      ]
```

**Review by danranVm — CHANGES_REQUESTED** [review]

File: packages/components/spin/style/index.less [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  bottom: 0;
+  user-select: none;
+}
+
+.@{spin-prefix} {
+  position: relative;
+
+  &-spinner {
+    @spin-mask();
+    z-index: 12;
```
Comment (Usualminds) [inline packages/components/spin/style/index.less:19]: ` z-index: 12` 建议改为 10 整数

**Review by Usualminds — COMMENTED** [review]
@liuzaijiang 没什么问题，层级那个你确认下是否需要修改

**Review by danranVm — APPROVED** [review]

**Review by Usualminds — APPROVED** [review]

**Review by danranVm — APPROVED** [review]

**Comment by danranVm** [comment]
@liuzaijiang 感谢你的 PR， LGTM.

**Pull request merged**